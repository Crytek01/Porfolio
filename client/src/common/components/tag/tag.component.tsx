import { FunctionComponent } from 'react';
import { theme } from '@infrastructure/theme';

import { StyledTag } from './tag.styles';

export interface ITag {
  textColor?: string;
  backgroundColor?: string;
  children: React.ReactNode;
}

export const Tag: FunctionComponent<ITag> = ({
  textColor = theme.colors.text.boldGrey,
  backgroundColor = theme.colors.backgroudColors.lightGreen,
  children,
}) => (
  <StyledTag textColor={textColor} backgroundColor={backgroundColor}>
    {children}
  </StyledTag>
);
