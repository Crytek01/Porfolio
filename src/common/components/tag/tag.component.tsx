import styled from 'styled-components';

interface ITag {
  textColor?: string;
  backgroundColor?: string;
}

export const Tag = styled.span<ITag>`
  color: #333;
  background-color: #c6f6d5;

  font-size: ${({ theme: { fontSize } }) => fontSize[1]};
  font-weight: ${({ theme: { fontWeight } }) => fontWeight.semibold};

  display: inline-block;
  text-align: center;

  padding: 0.25rem 0.5rem;

  text-transform: uppercase;
`;
