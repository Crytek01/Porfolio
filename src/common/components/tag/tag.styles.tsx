import styled from 'styled-components';


export interface ITagStyle {
  $textColor?: string;
  $backgroundColor?: string;
}

export const StyledTag = styled.span<ITagStyle>`
  color: ${({ $textColor }) => $textColor};
  background-color: ${({ $backgroundColor }) => $backgroundColor};

  font-size: ${({ theme: { fontSize } }) => fontSize[1]};
  font-weight: ${({ theme: { fontWeight } }) => fontWeight.semibold};

  display: inline-block;
  text-align: center;

  padding: 0.25rem 0.5rem;

  text-transform: uppercase;
`;
