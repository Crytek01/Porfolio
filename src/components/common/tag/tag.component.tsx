import styled from 'styled-components';

interface ITag {
  textColor?: string;
  backgroundColor?: string;
}

export const Tag = styled.span<ITag>`
  color: #333;
  background-color: #c6f6d5;

  font-size: 1.2rem;
  font-weight: 600;

  display: inline-block;
  text-align: center;

  padding: 0.25rem 0.5rem;

  text-transform: uppercase;
`;
