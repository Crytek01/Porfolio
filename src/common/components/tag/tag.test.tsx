import { shallowRender } from '@common/utils/test/shallowRender';
import { describe, expect, it } from 'vitest';

import { Tag } from './tag.component';

describe('Tag component Test', () => {
  it('renders correctly', () => {
    const renderer = shallowRender(
      <Tag backgroundColor="#555" textColor="#484848">
        Tag
      </Tag>
    );

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
