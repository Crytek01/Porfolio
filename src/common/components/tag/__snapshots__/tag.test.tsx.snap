// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tag component Test > renders correctly 1`] = `
<styled.span
  $backgroundColor="#555"
  $textColor="#484848"
>
  Tag
</styled.span>
`;
